### ISSUE TITLE
new `alias` does not register usage

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#8678]
### Describe the bug

with the new `alias` implementation, the comment before `alias`es is not treated as the usage anymore...

### How to reproduce

1. run the following in the REPL
```bash
# usage for foo
alias foo = echo "foo"

# usage for bar
old-alias bar = echo "bar"
```
2. try the aliases => they both work
```bash
>_ foo
foo
>_ bar
bar
```
3. but the usage is not registered for the `foo` alias
```bash
>_ $nu.scope.aliases | where name == foo | select name usage
╭───┬──────┬───────╮
│ # │ name │ usage │
├───┼──────┼───────┤
│ 0 │ foo  │       │
╰───┴──────┴───────╯
```
4. and it is with `bar`
```bash
>_ $nu.scope.aliases | where name == bar | select name usage
╭───┬──────┬───────────────╮
│ # │ name │     usage     │
├───┼──────┼───────────────┤
│ 0 │ bar  │ usage for bar │
╰───┴──────┴───────────────╯
```

### Expected behavior

with the same example, i expected the `bar` alias to behave the same and the `foo` alias to behave as the `bar` alias
```bash
>_ $nu.scope.aliases | where name == foo | select name usage
╭───┬──────┬───────────────╮
│ # │ name │     usage     │
├───┼──────┼───────────────┤
│ 0 │ foo  │ usage for foo │
╰───┴──────┴───────────────╯
```

### Configuration

| key                | value                                               |
| ------------------ | --------------------------------------------------- |
| version            | 0.77.2                                              |
| branch             | feature/implement-help-commands-in-standard-library |
| commit_hash        | 6f7c7a18d1e5128d8c3ab6d7a7054f6268c598c5            |
| build_os           | linux-x86_64                                        |
| build_target       | x86_64-unknown-linux-gnu                            |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)                 |
| rust_channel       | 1.66.1-x86_64-unknown-linux-gnu                     |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)                 |
| build_time         | 2023-03-29 19:32:23 +02:00                          |
| build_rust_channel | release                                             |
| features           | default, zip                                        |
| installed_plugins  | gstat                                               |

**Comment by fdncred** [comment]
Just to pile on here, what's weirder to me is that if your alias references a custom command, the custom command's help is listed.
```
help aliases 
╭───┬──────┬─────────────────────────┬───────────────────────────────────────────────╮
│ # │ name │        expansion        │                     usage                     │
├───┼──────┼─────────────────────────┼───────────────────────────────────────────────┤
│ 0 │ z    │ __zoxide_z              │ Jump to a directory using only keywords.      │
│ 1 │ zi   │ __zoxide_zi             │ Jump to a directory using interactive search. │
```

And these aliases are declared like this
```
alias z = __zoxide_z
alias zi = __zoxide_zi
```
and point to these custom commands
```
# Jump to a directory using only keywords.
def-env __zoxide_z [...rest:string] {
  # `z -` does not work yet, see https://github.com/nushell/nushell/issues/4769
  let arg0 = ($rest | append '~').0
  let path = if (($rest | length) <= 1) and ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {
    $arg0
  } else {
    (zoxide query --exclude $env.PWD -- $rest | str trim -r -c "\n")
  }
  cd $path
}

# Jump to a directory using interactive search.
def-env __zoxide_zi  [...rest:string] {
  cd $'(zoxide query -i -- $rest | str trim -r -c "\n")'
}
```
kind of confusing

**Comment by kubouch** [comment]
Yeah, there might be some weird behavior that we might need to iron out. Basically all calls of the Command trait of an alias are passed to the wrapped commands. So when you call `usage()` of the `foo` alias (e.g., by `help foo`), it will call `usage()` of the `echo` command. I did it that way for simplicity but it might not be optimal.

**Comment by alexjp** [comment]
> Yeah, there might be some weird behavior that we might need to iron out

maybe this isn't the right place, but some weird behavior:

```
alias test = echo hi
alias test = test world

test
───┬───────╮
│ 0 │ hi    │
│ 1 │ world │
╰───┴───────╯


$nu.scope.aliases | where name == test
╭───┬──────┬────────────┬───────╮
│ # │ name │ expansion  │ usage │
├───┼──────┼────────────┼───────┤
│ 0 │ test │ test world │       │
╰───┴──────┴────────────┴───────╯
```


its nice that the new alias in 0.78 allows to "build upon" previous alias :P but shouldn't it show in the `$nu.scope.aliases | where name == test` ? (meaning, both `test = echo hi` and `test = test world`)

**Comment by kubouch** [comment]
It shouldn't show both because only the last one is callable, but it should probably show the full expansion.

**Comment by alexjp** [comment]
> It shouldn't show both because only the last one is callable, but it should probably show the full expansion.

maybe if the full expansion is too tricky... just a note/symbol/info that is "being composed upon others" would be enough for me. :)

**Comment by amtoine** [comment]
i can still reproduce this, the only change is that `old-alias` does not exist anymore.

**Comment by weirdan** [comment]
It works now:

```console
$ # usage for foo
  alias foo = echo "foo"

$ scope aliases | where name == foo | select name expansion usage
╭───┬──────┬────────────┬───────────────╮
│ # │ name │ expansion  │     usage     │
├───┼──────┼────────────┼───────────────┤
│ 0 │ foo  │ echo "foo" │ usage for foo │
╰───┴──────┴────────────┴───────────────╯
```
